### ISSUE TITLE
jnius.detach() not safe with  local variables

### ISSUE THREAD

**Issue by yasiha** [kivy/pyjnius#240]
Sorry, my English is not good, Can only give examples：

ByteBuffer = autoclass('java.nio.ByteBuffer')
Thread(target = runthread).start()
def runthread():
    try:
        dst = ByteBuffer.allocate(20)
        .....
    finnaly:
        jnius.detach()

The program crashes when exiting the thread, Unless we do things like this

def runthread():
    dst = None
    try:
        dst = ByteBuffer.allocate(20)
        .....
    finnaly:
        if dst: del dst             #dealloc in current jnienv
        jnius.detach()

so why not give us  jnius.thread

I hope someone can understand what I said

**Comment by akshayaurora** [comment]
This should now work as is without you having to call detach() manually https://github.com/kivy/pyjnius/commit/b5a1ecb7ec409c37e955acd6a12eb1d75ba3d1ab
